src/test/java/namor/template/calculator/RPNcalculatorTest.java

```diff
@@ -64,4 +66,42 @@ public void whenTwoArithmaticOperatinsDoneReurnsCorrectNumber() {
         assertThat(actual).isEqualTo(expected);
     }
 
+    @Test
+    public void stackWorksWithFloat() {
+        RPNcalculator<Float> FrpnCalculator = new RPNcalculator<>(Float.class, ArrayStack.class);
+        final float expected = 4.4f;
+        final float actual;
+        final String arithmeticExpression = "2.2 3.3 + 1.1 -";
+
+        actual = FrpnCalculator.calculate(arithmeticExpression);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void stackWorksWithDouble() {
+        RPNcalculator<Double> FrpnCalculator = new RPNcalculator<>(Double.class, ArrayStack.class);
+        final double expected = 4.4;
+        final double actual;
+        final String arithmeticExpression = "2.2 3.3 + 1.1 -";
+
+        actual = FrpnCalculator.calculate(arithmeticExpression);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void stackWorksWithLong() {
+        RPNcalculator<Long> FrpnCalculator = new RPNcalculator<>(Long.class, ArrayStack.class);
+        final long expected = 4L;
+        final long actual;
+        final String arithmeticExpression = "2 3 + 1 -";
+
+        actual = FrpnCalculator.calculate(arithmeticExpression);
+        assertThat(actual).isEqualTo(expected);
+    }
+    
+    @Test
+    public void stackWorksWithMapBasedStack() {
+        RPNcalculator<Integer> mrpnCalculator = new RPNcalculator<>(Integer.class, MapBasedStack.class);
+        assertThat(mrpnCalculator.getNumbers().getClass()).isEqualTo(MapBasedStack.class);
+    }
 }
```

Comment:
You shouldn't check things which java does, only logic of your code. So don't write tests if generics works correct or java know how to deal with Double type.